subsys/net/lib/sockets/sockets_tls.c

```diff
@@ -323,6 +323,52 @@ static inline bool is_handshake_complete(struct tls_context *ctx)
 	return k_sem_count_get(&ctx->tls_established) != 0;
 }
 
+#if defined(CONFIG_NET_SOCKETS_TLS_SET_MAX_FRAGMENT_LENGTH) &&	\
+	defined(MBEDTLS_SSL_MAX_FRAGMENT_LENGTH) &&		\
+	(MBEDTLS_TLS_EXT_ADV_CONTENT_LEN < 16384)
+/*
+ * Copied from include/mbedtls/ssl_internal.h
+ *
+ * Maximum length we can advertise as our max content length for
+ * RFC 6066 max_fragment_length extension negotiation purposes
+ * (the lesser of both sizes, if they are unequal.)
+ */
+#define MBEDTLS_TLS_EXT_ADV_CONTENT_LEN (                            \
+	(MBEDTLS_SSL_IN_CONTENT_LEN > MBEDTLS_SSL_OUT_CONTENT_LEN)   \
+	? (MBEDTLS_SSL_OUT_CONTENT_LEN)				     \
+	: (MBEDTLS_SSL_IN_CONTENT_LEN)				     \
+	)
+
+BUILD_ASSERT(MBEDTLS_TLS_EXT_ADV_CONTENT_LEN >= 512,
+	     "Too small content length!");
+
+static inline unsigned char mbedtls_mfl_code_from_content_len(void)
+{
+	size_t len = MBEDTLS_TLS_EXT_ADV_CONTENT_LEN;
+
+	if (len >= 4096) {
+		return MBEDTLS_SSL_MAX_FRAG_LEN_4096;
+	} else if (len >= 2048) {
+		return MBEDTLS_SSL_MAX_FRAG_LEN_2048;
+	} else if (len >= 1024) {
+		return MBEDTLS_SSL_MAX_FRAG_LEN_1024;
+	} else if (len >= 512) {
+		return MBEDTLS_SSL_MAX_FRAG_LEN_512;
+	} else {
+		return MBEDTLS_SSL_MAX_FRAG_LEN_INVALID;
+	}
+}
+
+static inline void mbedtls_set_max_frag_len(mbedtls_ssl_config *config)
```

Comment:
I'm not sure about these `mbedtls_` prefixes for functions here - so far this prefix has been reserved for mbedtls API, so it might be confusing for users. How about `tls_` prefix, just as other functions in this file?